library/hmac_drbg.c

```diff
@@ -390,6 +429,21 @@ int mbedtls_hmac_drbg_random_with_add( void *p_rng,
 
 exit:
     /* 8. Done */
+
+    if (ret == 0)
```

Comment:
The checking of doubles and illegal conditions can now be omitted by skipping this branching. It would be better to have a `ret_fi` -variable which is initially `MBEDTLS_ERR_PLATFORM_FAULT_DETECTED` and set to 0 when everything is Ok and returned in the end.